### PR TITLE
Remove sender extraction http requests in RasaChatInput

### PR DESCRIPTION
**Pull request by ricwo**
**Proposed changes**:
- fixes https://github.com/RasaHQ/rasa-x/issues/1113

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/channels/rasa_chat.py [hunk 1]
```diff
@@ -27,35 +28,58 @@ def from_credentials(cls, credentials):
...
+                    logger.debug(
+                        "Fetched JWT public key for algorithm '{}':\n{}"
+                        "".format(self.jwt_algorithm, self.jwt_key)
                     )
-                    return None
+                else:
+                    logger.info("Could not find JWT public key at `/version` endpoint.")
+
+    async def _decode_jwt(self, bearer_token: Text) -> Dict:
+        bearer_token_prefix = "Bearer "
```
Comment (tabergma) [inline rasa/core/channels/rasa_chat.py]: Can we make this a constant?

File: rasa/core/channels/rasa_chat.py [hunk 2]
```diff
@@ -27,35 +28,58 @@ def from_credentials(cls, credentials):
...
+                public_key_url, timeout=DEFAULT_REQUEST_TIMEOUT
             ) as resp:
-                if resp.status == 200:
-                    return await resp.json()
-                else:
-                    logger.info(
-                        "Failed to check token: {}. "
-                        "Content: {}".format(token, await resp.text())
+                if resp.status != 200:
+                    logger.info("Failed to fetch ")
```
Comment (tabergma) [inline rasa/core/channels/rasa_chat.py]: Can you make the log message more informative? Something like "Failed to fetch public key from 'url'. Response: ...."

File: rasa/core/channels/rasa_chat.py (hunk 1 above)
Comment (tabergma) [inline rasa/core/channels/rasa_chat.py]: Please, use a constant here.

File: rasa/core/channels/rasa_chat.py [hunk 3]
```diff
@@ -27,35 +28,58 @@ def from_credentials(cls, credentials):
...
+
+        # noinspection PyBroadException
+        try:
+            return await self._decode_jwt(bearer_token)
+        except jwt.exceptions.InvalidSignatureError:
+            logger.error("JWT public key invalid, fetching new one.")
+            await self._fetch_public_key()
+            return await self._decode_jwt(bearer_token)
+        except Exception:
+            logger.exception("Failed to decode bearer token.")
```
Comment (tabergma) [inline rasa/core/channels/rasa_chat.py]: Can you please also log the exception. Maybe just as debug.

File: rasa/core/channels/rasa_chat.py [hunk 4]
```diff
@@ -27,35 +28,58 @@ def from_credentials(cls, credentials):
...
+                    self.jwt_algorithm = rjs["keys"][0]["alg"]
+                    logger.debug(
+                        "Fetched JWT public key for algorithm '{}':\n{}"
+                        "".format(self.jwt_algorithm, self.jwt_key)
                     )
-                    return None
+                else:
+                    logger.info("Could not find JWT public key at `/version` endpoint.")
+
+    async def _decode_jwt(self, bearer_token: Text) -> Dict:
```
Comment (tabergma) [inline rasa/core/channels/rasa_chat.py]: We can remove the `async`.

File: rasa/core/channels/rasa_chat.py (hunk 3 above)
Comment (ricwo) [reply rasa/core/channels/rasa_chat.py]: will be logged automatically with `logger.exception`

File: rasa/core/channels/rasa_chat.py (hunk 1 above)
Comment (ricwo) [reply rasa/core/channels/rasa_chat.py]: 👍

**Pull request merged**